File: quarkus/fta/src/main/java/fta/entity/StreetTrafficReport.java
package fta.entity;

import io.quarkus.mongodb.panache.PanacheMongoEntity;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class StreetTrafficReport extends PanacheMongoEntity {
    private String streetUUID;
    private int sumTraffic;
    private int maxTraffic;
    private int minTraffic;
    private double avgTraffic;
    private double avgVelocity;
    private double maxVelocity;
}

File: ftd_webapp/lib/features/data/models/street_traffic_report/street_traffic_report.dart
// ignore_for_file: invalid_annotation_target

import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:flutter/foundation.dart';

part 'street_traffic_report.freezed.dart';
part 'street_traffic_report.g.dart';

@freezed
class StreetTrafficReport with _$StreetTrafficReport{
  factory StreetTrafficReport({
    required String streetUUID,
    required double maxTraffic,
    required double minTraffic,
    required double avgTraffic,
    required double sumTraffic,
    required double maxVelocity,
    required double avgVelocity
  }) = _StreetTrafficReport;

  factory StreetTrafficReport.fromJson(Map<String, dynamic> json) => _$StreetTrafficReportFromJson(json);

  factory StreetTrafficReport.empty(String uuid) => StreetTrafficReport(
    streetUUID: uuid,
    maxTraffic: 0,
    minTraffic: 0,
    avgTraffic: 0,
    sumTraffic: 0,
    maxVelocity: 0,
    avgVelocity: 0
  );

}